spring/src/main/java/io/seata/spring/annotation/GlobalTransactionScanner.java

```diff
@@ -469,8 +480,71 @@ public void afterPropertiesSet() {
         if (initialized.compareAndSet(false, true)) {
             initClient();
         }
+
+        this.findBusinessBeanNamesNeededEnhancement();
     }
 
+    private void findBusinessBeanNamesNeededEnhancement() {
+        if (applicationContext instanceof ConfigurableApplicationContext) {
+            ConfigurableApplicationContext configurableApplicationContext = (ConfigurableApplicationContext) applicationContext;
+            ConfigurableListableBeanFactory configurableListableBeanFactory = configurableApplicationContext.getBeanFactory();
+
+            String[] beanNames = applicationContext.getBeanDefinitionNames();
+            for (String contextBeanName : beanNames) {
+                BeanDefinition beanDefinition = configurableListableBeanFactory.getBeanDefinition(contextBeanName);
+                if (StringUtils.isBlank(beanDefinition.getBeanClassName())) {
+                    continue;
+                }
+                if (IGNORE_ENHANCE_CHECK_SET.contains(beanDefinition.getBeanClassName())) {
+                    continue;
+                }
+                try {
+                    // get the class by bean definition class name
+                    Class<?> beanClass = Class.forName(beanDefinition.getBeanClassName());
+                    // check if it needs enhancement by the class
+                    IfNeedEnhanceBean ifNeedEnhanceBean = DefaultInterfaceParser.get().parseIfNeedEnhancement(beanClass);
+                    if (!ifNeedEnhanceBean.isIfNeed()) {
+                        continue;
+                    }
+                    if (ifNeedEnhanceBean.getNeedEnhanceEnum().equals(NeedEnhanceEnum.SERVICE_BEAN)) {
+                        // the native bean which dubbo, sofa bean service bean referenced
+                        PropertyValue propertyValue = beanDefinition.getPropertyValues().getPropertyValue("ref");
+                        if (propertyValue == null) {
```

Comment:
这些逻辑看起来应该和handler强相关，是不是该挪到handler去？虽然这样做会带来一部分重复的代码
This logic seems like it should be strongly related to the handler, so should it be moved to the handler, even though it would be a duplicate piece of code?

Reply:
如果引入到parser中，可能会给parser带来beanDefinition的spring依赖，原则上可能不想让parser层引入spring相关的依赖